packages/pluggableWidgets/screenshotrunner-native/src/ScreenshotRunner.tsx

```diff
@@ -0,0 +1,158 @@
+            masterIsRunning: boolean;
+            task?: {
+                name: string;
+                isRunning: boolean;
+                error: boolean;
+                onError: (e: Error, name: string) => any;
+            };
+        };
+    };
+
+const timeout = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+export function ScreenshotRunner(props: ScreenshotRunnerProps<ScreenshotRunnerStyle>): ReactElement {
+    const [tasksRunning, setTasksRunning] = useState<TasksType[]>([]);
+    const [taskWithErrors, setTaskWithErrors] = useState<ErrorType[]>([]);
+    const [areTasksTriggered, setAreTasksTriggered] = useState(false);
+    const styles = flattenStyles(defaultScreenshotRunnerStyles, props.style);
+    const { androidRipple: androidRippleStart } = styles.button.start.container;
+    const { androidRipple: androidRippleStop } = styles.button.stop.container;
+    const tasksEnabledLength = props.tasks.filter(task => task.enabled).length;
+
+    const onError = (e: Error, name: string) => {
+        console.error(e);
+        setTaskWithErrors(oldArray => [
+            ...oldArray,
+            {
+                name,
+                error: e
+            }
+        ]);
+    };
+    const enableGlobalScreenshotVariable = (name = "placeholder") => {
+        (global as CustomGlobal).screenshotRunner = {
+            masterIsRunning: true,
+            task: {
+                name,
+                isRunning: true,
+                error: false,
+                onError: (e: Error) => {
+                    onError(e, name);
+                }
+            }
+        };
+    };
+
+    const runTask = async (task: TasksType) => {
+        const maxTimeouts = 30;
+        let timeoutCount = 0;
+
+        if (!task.task?.canExecute) {
+            throw new Error(`Task ${task.name} is not executable..`);
+        }
+
+        setTasksRunning(oldArray => [...oldArray, task]);
+        enableGlobalScreenshotVariable(task.name);
+        executeAction(task.task);
+
+        while ((global as CustomGlobal).screenshotRunner?.task?.isRunning && timeoutCount <= maxTimeouts) {
+            console.info("Waiting for task to complete..");
+            await timeout(1000);
+            timeoutCount++;
+        }
+        const g = global as CustomGlobal;
+        if (timeoutCount < maxTimeouts && !g.screenshotRunner?.task?.error) {
+            console.info("Task Completed!");
+        } else if (g.screenshotRunner?.task?.error) {
+            console.error("Task Failed! Capturing screenshot triggered an error.");
+        } else {
+            console.error("Task Failed! Timeout has been reached.");
+        }
+    };
+
+    useEffect(() => {
+        if (areTasksTriggered) {
+            (async () => {
+                for await (const task of props.tasks) {
+                    if ((global as CustomGlobal).screenshotRunner?.masterIsRunning && task.enabled) {
+                        await runTask(task);
```

Comment:
Probably this code will miss properties update due  to its wrapped inside a closure but without updates for the props (bind).